python_modules/libraries/dagster-gcp-pyspark/dagster_gcp_pyspark/bigquery/bigquery_pyspark_type_handler.py

```diff
@@ -0,0 +1,161 @@
+            from dagster import asset, Definitions
+
+            @asset(
+                key_prefix=["my_dataset"]  # will be used as the dataset in BigQuery
+            )
+            def my_table():
+                ...
+
+            bigquery_io_manager = build_bigquery_io_manager([BigQueryPySparkTypeHandler()])
+
+            defs = Definitions(
+                assets=[my_table],
+                resources={
+                    "io_manager": bigquery_io_manager.configured({
+                        "project" : {"env": "GCP_PROJECT"}
+                    })
+                }
+            )
+
+    """
+
+    def handle_output(
+        self, context: OutputContext, table_slice: TableSlice, obj: DataFrame, _
+    ) -> Mapping[str, RawMetadataValue]:
+        options = _get_bigquery_write_options(context.resource_config, table_slice)
+
+        with_uppercase_cols = obj.toDF(*[c.upper() for c in obj.columns])
+
+        with_uppercase_cols.write.format("bigquery").options(**options).mode("append").save()
+
+        return {
+            "dataframe_columns": MetadataValue.table_schema(
+                TableSchema(
+                    columns=[
+                        TableColumn(name=field.name, type=field.dataType.typeName())
+                        for field in obj.schema.fields
+                    ]
+                )
+            ),
+        }
+
+    def load_input(self, context: InputContext, table_slice: TableSlice, _) -> DataFrame:
+        options = _get_bigquery_read_options(table_slice)
+        spark = SparkSession.builder.getOrCreate()
+
+        df = (
+            spark.read.format("bigquery")
+            .options(**options)
+            .load(BigQueryClient.get_select_statement(table_slice))
+        )
+
+        return df.toDF(*[c.lower() for c in df.columns])
+
+    @property
+    def supported_types(self):
+        return [DataFrame]
+
+
+bigquery_pyspark_io_manager = build_bigquery_io_manager(
+    [BigQueryPySparkTypeHandler()], default_load_type=DataFrame
+)
+bigquery_pyspark_io_manager.__doc__ = """
+An IO manager definition that reads inputs from and writes PySpark DataFrames to BigQuery.
+
+Returns:
+    IOManagerDefinition
+
+Examples:
+
+    .. code-block:: python
+
+        from dagster_bigquery_pyspark import bigquery_pyspark_io_manager
+        from dagster import Definitions
+
+        @asset(
+            key_prefix=["my_dataset"]  # will be used as the dataset in BigQuery
+        )
+        def my_table() -> pd.DataFrame:  # the name of the asset will be the table name
+            ...
+
+        defs = Definitions(
+            assets=[my_table],
+            resources={
+                "io_manager": bigquery_pyspark_io_manager.configured({
+                    "project" : {"env": "GCP_PROJECT"}
+                })
+            }
+        )
+
+    You can tell Dagster in which dataset to create tables by setting the "dataset" configuration value.
```

Comment:
Should this be 'the "schema" metadata value'?

Reply:
it's `dataset` when it's configured on the io manager itself. The only time it's `schema` is for output metadata on ops